### DISCUSSION TITLE
Question: how to paste to clipboard over SSH using hints?

### DISCUSSION THREAD

**Discussion by rachel-1** [kovidgoyal/kitty#5681] [Q&A] [answered]
I have a simple `custom_hints.py` (based on [this page of the docs](https://sw.kovidgoyal.net/kitty/kittens/hints/#completely-customizing-the-matching-and-actions-of-the-kitten)) I created to paste lines to the clipboard (which accounts for multilines and how Emacs adds special characters for line-wrapping). However, I can't get the same clipboard functionality as using "--program @". I have gotten pasting to the clipboard to work locally, but not over SSH.

`custom_hints.py`:
```
import re

def mark(text, args, Mark, extra_cli_args, *a):
    # This function is responsible for finding all
    # matching text. extra_cli_args are any extra arguments
    # passed on the command line when invoking the kitten.
    # We mark all individual word for potential selection
    for idx, m in enumerate(re.finditer(r'^.*?(\S+[/|\.].*?)\$', text, re.MULTILINE | re.DOTALL)):
        start, end = m.span(1)
        mark_text = text[start:end].replace('\\', '').replace('\n', '')
        # The empty dictionary below will be available as groupdicts
        # in handle_result() and can contain arbitrary data.
        yield Mark(idx, start, end, mark_text, {})


def handle_result(args, data, target_window_id, boss, extra_cli_args, *a):
    matches, groupdicts = [], []
    for m, g in zip(data['match'], data['groupdicts']):
        if m:
            matches.append(m), groupdicts.append(g)
    for word, match_data in zip(matches, groupdicts):
        w = boss.window_id_map.get(target_window_id)
        if w is not None:
            boss.clipboard.set_text(word)
```

`kitty.conf`:
```
map f1 kitten hints --customize-processing custom_hints.py
map f2 kitten hints --type word --program @    
```

On version 0.26.5, F2 will copy to the clipboard even over SSH, but F1 only works locally (not over SSH).

**Comment by page-down** [comment, marked as answer]
A better place to ask questions is [GitHub Discussions](https://github.com/kovidgoyal/kitty/discussions).

Since hints works at the terminal level, any text displayed on the screen can be copied to the local clipboard.
This is not affected by whatever program (including SSH clients) is running.

Using the official example in the documentation, calling `boss.clipboard.set_text(word)`, will copy the selected word to the clipboard and it works fine after connecting to ssh.

**Reply by rachel-1** [reply]
Thanks! I got it working after rebooting Kitty and fiddling with the way I was pasting into Emacs. It was very helpful to know that I was closer than I had thought.
